src/bin/pg_autoctl/cli_show.c

```diff
@@ -1490,8 +1481,27 @@ fprint_pidfile_as_json(const char *pidfile)
 			continue;
 		}
 
+		/* data directory */
+		if (pidLine == PIDFILE_LINE_DATA_DIR)
+		{
+			json_object_set_string(jsobj, "pgdata", fileLines[lineNumber]);
+		}
+
+		/* version string */
+		if (pidLine == PIDFILE_LINE_VERSION_STRING)
+		{
+			json_object_set_string(jsobj, "version", fileLines[lineNumber]);
+		}
+
+		/* extension version string */
+		if (pidLine == PIDFILE_LINE_EXTENSION_VERSION)
+		{
+			/* skip it, the supervisor does not connect to the monitor */
+			(void) 0;
+		}
+
 		/* semId */
-		if (lineNumber == 1)
+		if (pidLine == PIDFILE_LINE_SEM_ID)
 		{
 			int semId = 0;
 			stringToInt(fileLines[1], &semId);
```

Comment:
`semId` in JSON is `0` for me. I guess because this call fails and we ignore it. The actual ID is `1437007872`, which might be helpfull in debugging why. So two things should be changed here:
1. handle error with `stringToInt`.
2. Fix the underlying issue.

Maybe this is already enough to fix the issue:
```suggestion
			stringToInt(fileLines[lineNumber], &semId);
```

Reply:
Fixed now, thanks.